packages/react/src/components/Popover/index.tsx

```diff
@@ -0,0 +1,283 @@
+      setTargetElement(targetElement);
+    }, [target]);
+
+    useEffect(() => {
+      return () => {
+        isolator?.deactivate();
+      };
+    }, []);
+
+    useEffect(() => {
+      if (!isolator) return;
+
+      if (show) isolator.activate();
+      else isolator.deactivate();
+    }, [show, isolator]);
+
+    useEffect(() => {
+      if (popoverRef.current) attachIsolator();
+    }, [popoverRef.current]);
+
+    useEffect(() => {
+      if (show && popoverRef.current) {
+        // Find the first focusable element inside the container
+        const firstFocusableElement = popoverRef.current.querySelector(
+          focusableSelector
+        );
+
+        if (firstFocusableElement instanceof HTMLElement) {
+          firstFocusableElement.focus();
+        }
+      }
+
+      targetElement?.setAttribute('aria-expanded', Boolean(show).toString());
+    }, [show, popoverRef.current]);
+
+    useEffect(() => {
+      const handleEscape = (event: KeyboardEvent) => {
+        if (
+          event.key === 'Escape' ||
+          event.key === 'Esc' ||
+          event.keyCode === 27
+        ) {
+          handleClosePopover();
+        }
+      };
+
+      if (show) {
+        document.body.addEventListener('keyup', handleEscape);
+      } else {
+        document.body.removeEventListener('keyup', handleEscape);
+      }
+
+      return () => {
+        document.body.removeEventListener('keyup', handleEscape);
+      };
+    }, [show]);
+
+    useEffect(() => {
+      const attrText = targetElement?.getAttribute('aria-controls');
+      const hasPopupAttr = targetElement?.getAttribute('aria-haspopup');
+
+      if (!attrText?.includes(id) && show) {
+        targetElement?.setAttribute('aria-controls', id);
+      }
+
+      if (!hasPopupAttr) {
+        targetElement?.setAttribute('aria-haspopup', 'true');
+      }
+    }, [targetElement, id, show]);
+
+    const handleClickOutside = () => {
+      if (show) {
+        handleClosePopover();
+      }
+    };
+
+    const attachIsolator = () => {
+      if (popoverRef?.current) {
+        setIsolator(new AriaIsolate(popoverRef?.current));
+      }
+    };
+
+    const handleClosePopover = () => {
+      isolator?.deactivate();
+      if (show) {
+        onClose();
+      }
+    };
+
+    if (!show || !isBrowser()) return null;
+
+    return createPortal(
+      <FocusTrap
+        focusTrapOptions={{
+          allowOutsideClick: true,
+          fallbackFocus: '.Popover__borderLeft'
+        }}
+      >
+        <ClickOutsideListener onClickOutside={handleClickOutside}>
+          <div
+            id={id}
+            className={classnames(
+              'Popover',
+              `Popover--${placement}`,
+              className,
+              {
+                'Popover--hidden': !show,
+                'Popover--alert': variant === 'alert'
+              }
+            )}
+            ref={popoverRef}
+            role="dialog"
```

Comment:
Sorry I missed this earlier, but we need to set an accessible name for popover since it's a dialog. It's a bit tricky based on the current structure since we don't have direct access to the popover content.

One way we could do this is by passing an id into `AlertPopoverContent` which would get labelled by the `infoText`. For the custom side of things, we can just show an example of how to do this via documentation.